Lib/java/java.swg

```diff
@@ -1421,10 +1421,24 @@ SWIG_PROXY_CONSTRUCTOR(true, true, SWIGTYPE)
 %apply SWIGTYPE & { SWIGTYPE (CLASS::*const&) }
 
 /* String & length */
+%pragma(java) jniclasscode=%{
```

Comment:
This pragma has reappeared again, did you overlook my previous comment:


             Please apply this patch to your branch:
[bytes.patch.txt](https://github.com/swig/swig/files/13362083/bytes.patch.txt)

_Originally posted by @wsfulton in https://github.com/swig/swig/issues/2597#issuecomment-1812004120_

Reply:
And similar removal of the pragma is needed for C#

Reply:
> And similar removal of the pragma is needed for C#

The [bytes.patch.txt](https://github.com/swig/swig/files/13362083/bytes.patch.txt) patch  add a '\0' to `li_cdata_cpp_runme.*` and `li_cdata_c_runme.*`.
I did that to all languages that support including Java.

I did not see any patch/remark regarding the `%pragma` in java and C#.
How can I add functions to Java/C# without using `%pragma`?

Reply:
After changing cdate  back to byte array in  Java, I revert `li_cdata_cpp_runme.java` and `li_cdata_c_runme.java`  back, so I  rebased your patch into my single commit.
The `%pragma` was not duplicate. Simply in C# and Java, the cdata uses byte array while STRING uses string type. So each typemap need a separate C#/Java `%pragma`, there are no duplications.

Erez

Reply:
Oh dear, I was actually referring to an additional patch that I didn't provide, so I'm not surpised you're confused. It's this one: 
[remove-java-pragma.patch.txt](https://github.com/swig/swig/files/14364706/remove-java-pragma.patch.txt)
I wrote this a while ago, there is info in the commit message, but the idea is to remove the pragma which always generates a Java method which normally is not used. Can you see if you can incorporate this and an equivalent set of changes for C#?

Reply:
> Oh dear, I was actually referring to an additional patch that I didn't provide, so I'm not surpised you're confused. It's this one: [remove-java-pragma.patch.txt](https://github.com/swig/swig/files/14364706/remove-java-pragma.patch.txt) I wrote this a while ago, there is info in the commit message, but the idea is to remove the pragma which always generates a Java method which normally is not used. Can you see if you can incorporate this and an equivalent set of changes for C#?

I'll try.
If I fail, or have more issues, I will flag for help.

Thanks
   Erez

Reply:
In the case of C#.

I think although C# share some similarity with Java.
The C# to native is differ from JNI.

In our case we need to transfer the pointer and the length in a single object.

In `Lib/typemaps/cdata.swg` and `Lib/csharp/csharp.swg`.
The pragma is declaring the prototype of a C function called from C#.
There is a way to allocate C memory in C#, but I think this is a more simple approach.
[Marshalling Classes, Structures, and Unions](https://learn.microsoft.com/en-us/dotnet/framework/interop/marshalling-classes-structures-and-unions)

Anyhow, I think it is does require at least a C# functions in PINVOKE.

Yet, I am open for suggestions
    Erez